public/js/utils/editor.js

```diff
@@ -0,0 +1,32 @@
+const { isPretty } = require("./source");
```

Comment:
What' the point of creating another file? I don't see why we should be splitting out components into multiple files between `utils` and `components`. It's a lot hard to follow the codebase.